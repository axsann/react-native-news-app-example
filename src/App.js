// @flow

import React, { Component } from 'react'
import { Provider } from 'react-redux'
import store from './redux/Store'
import AppWithNavigationState from './navigators/AppNavigator'

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <AppWithNavigationState />
      </Provider>
    )
  }

}
