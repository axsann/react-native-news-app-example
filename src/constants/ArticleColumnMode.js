export default {
  TWO_COLUMN: 'TWO_COLUMN',
  THREE_COLUMN: 'THREE_COLUMN',
}
