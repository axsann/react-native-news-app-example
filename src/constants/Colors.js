export default {
  WHITE: '#ffffff',
  BLACK: '#000000',
  TRANSPARENT: 'transparent',
  BRIGHT_BLUE: '#387ef5',
  BRIGHT_RED: '#ef473a',
  LIME_GREEN: '#33cd5f',
  VERY_LIGHT_GRAY: '#eeeeee',
}
