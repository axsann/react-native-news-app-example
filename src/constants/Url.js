export default {
  JSON_URL: 'https://gist.githubusercontent.com/axsann/99a77c2f1dae5317915f401b6d24f623/raw/cce84bae5485ae362cb837d9192d747129f95184/news_sample_data.json',
}
