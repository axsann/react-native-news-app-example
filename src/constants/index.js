import _COLORS from './Colors'
import _ARTICLE_COLUMN_MODE from './ArticleColumnMode'
import _URL from './Url'

export const COLORS = _COLORS
export const ARTICLE_COLUMN_MODE = _ARTICLE_COLUMN_MODE
export const URL = _URL
