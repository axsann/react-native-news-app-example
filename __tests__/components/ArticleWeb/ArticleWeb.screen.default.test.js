// @flow

import 'react-native'
import React from 'react'
import { mock, release } from 'mock-async-storage'
import ArticleWebScreen from '../../../src/components/ArticleWeb/ArticleWeb.screen'
import { ARTICLE_FONT_SIZE } from '../../../src/constants'
import ShallowRenderer from 'react-test-renderer/shallow'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

mock()

import { AsyncStorage as storage } from 'react-native'

jest.unmock('ScrollView')
jest.mock('WebView', () => 'WebView')

const URL = 'http://example.com'

const navigation = {
  state: {
    params: {
      url: URL,
    },
  },
}

describe('ArticleWebScreen', () => {
  it('renders correctly', () => {
    const instance = renderer.create(
      <ArticleWebScreen navigation={navigation} />
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })

  it('読み込みエラー時', () => {
    const shallowRenderer = new ShallowRenderer()
    shallowRenderer.render(<ArticleWebScreen navigation={navigation} />)
    const wrapper = shallowRenderer.getMountedInstance()

    wrapper.onError()
    expect(wrapper).toMatchSnapshot()
  })
})
