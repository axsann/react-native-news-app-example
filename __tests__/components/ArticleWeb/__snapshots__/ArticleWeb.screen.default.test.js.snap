// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleWebScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <WebView
    onError={[Function]}
    onLoad={[Function]}
    source={
      Object {
        "uri": "http://example.com",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <RNSpinkit
        color="#eeeeee"
        size={100}
        style={
          Array [
            Object {
              "height": 100,
              "width": 100,
            },
            Object {
              "marginTop": -90,
            },
          ]
        }
        type="ThreeBounce"
      />
    </View>
  </View>
</View>
`;

exports[`ArticleWebScreen 読み込みエラー時 1`] = `
ArticleWebScreen {
  "context": Object {},
  "onError": [Function],
  "props": Object {
    "navigation": Object {
      "state": Object {
        "params": Object {
          "url": "http://example.com",
        },
      },
    },
  },
  "refs": Object {},
  "state": Object {
    "isLoaded": true,
  },
  "updater": Updater {
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <ArticleWebScreen
        navigation={
                Object {
                        "state": Object {
                          "params": Object {
                            "url": "http://example.com",
                          },
                        },
                      }
        }
/>,
      "_instance": [Circular],
      "_newState": Object {
        "isLoaded": true,
      },
      "_rendered": <View
        style={
                Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "transparent",
                        "flex": 1,
                      }
        }
>
        <WebView
                onError={[Function]}
                onLoad={[Function]}
                source={
                        Object {
                                "uri": "http://example.com",
                              }
                }
                style={
                        Object {
                                "alignSelf": "stretch",
                                "flex": 1,
                              }
                }
        />
</View>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;
